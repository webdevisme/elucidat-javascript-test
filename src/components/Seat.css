.Seat {
    position: relative;

}
.Seat img {
    width: 30px;
    height: 20px;
}
.tooltip {
    display: none;
}
.Seat:hover .tooltip {
    display: inline-block;
}