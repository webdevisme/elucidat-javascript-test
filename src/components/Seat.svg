<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 1000 1000" enable-background="new 0 0 1000 1000" xml:space="preserve">
    <metadata> Svg Vector Icons : http://www.onlinewebfonts.com/icon </metadata>
    <g><g transform="translate(0.000000,511.000000) scale(0.100000,-0.100000)"><path d="M3339.7,4930.5c-631-91.6-1200.2-564.2-1418-1180.4c-96.5-274.7-113.8-514.7-113.8-1737.2v-1126h-277.2c-165.8,0-329.1-14.8-403.4-34.6c-348.9-99-722.6-383.6-873.5-665.7C1.4-286,63.3-850.2,412.2-1248.7C471.6-1318,578-1407,644.9-1449.1l123.7-76.7V-2236c0-640.9,5-722.6,47-814.2c94.1-207.9,314.3-363.8,556.8-395.9c61.9-9.9,113.8-24.7,113.8-34.6c0-12.4-66.8-202.9-148.5-428.1c-81.7-222.7-148.5-435.5-148.5-472.7c0-103.9,74.2-237.6,158.4-289.5c123.7-74.2,217.8-81.7,344-27.2c138.6,61.9,150.9,86.6,391,742.4l183.1,499.9h2734.5h2734.5l198-549.4c215.3-591.4,257.4-663.2,428.1-715.2c200.5-59.4,415.7,84.1,443,294.5c9.9,71.8-24.7,190.6-146,527.1l-158.4,438l118.8,12.4c240.1,29.7,452.9,175.7,556.8,383.6c39.6,81.7,44.6,173.2,52,816.6l7.4,725.1l121.3,74.2c324.2,198,544.4,618.7,544.4,1036.9c0,334.1-116.3,613.7-368.7,873.5c-297,309.3-707.8,462.8-1158.1,430.6l-205.4-14.9v1269.5c0,1180.4-2.5,1281.9-49.5,1460c-79.2,311.8-225.2,564.2-470.2,806.7c-309.3,309.3-603.8,460.3-1004.7,519.7C6438,4962.7,3542.6,4960.2,3339.7,4930.5z M6584,4215.3c452.9-86.6,806.7-462.8,866.1-920.6c12.4-99,24.7-767.1,24.7-1484.8V508.3l-123.7-146c-217.8-249.9-420.7-749.8-420.7-1034.4v-61.9H5000.2H3070v61.9c0,287-205.4,794.4-418.2,1031.9l-126.2,138.6v1333.8c0,1403.1,9.9,1561.5,111.4,1784.2c126.2,272.2,460.3,539.5,754.8,596.4C3550.1,4245,6423.1,4245,6584,4215.3z M1758.4,139.6c188.1-61.9,373.7-237.5,487.5-470.2c89.1-180.6,91.5-195.5,101.5-499.9c5-170.7,19.8-341.5,32.2-376.2c37.1-96.5,143.5-188.1,257.4-217.8c138.6-37.1,4588-37.1,4726.6,0c54.4,14.9,126.2,52,155.9,79.2c108.9,99,128.7,170.8,128.7,462.8c0,413.3,99,655.8,356.4,883.5c274.7,242.5,702.8,240,975-5c155.9-141.1,212.8-277.2,198-470.2c-29.7-366.2-378.6-546.9-640.9-334.1c-160.9,128.7-279.7,143.5-450.4,56.9c-160.9-81.7-220.3-341.5-108.9-502.3c69.3-103.9,257.4-232.6,415.7-289.5l121.3-44.5v-574.1v-576.6l-3518.9,5l-3521.4,7.4l-7.4,445.4c-2.5,245,0,502.3,7.4,569.2c14.8,123.7,17.3,126.2,133.6,168.3c158.4,56.9,346.5,188.1,418.2,289.5c108.9,160.8,49.5,420.7-111.4,502.3c-170.8,86.6-289.5,71.8-450.4-56.9C1201.6-1021,852.7-840.3,823-474.1c-14.9,193,42.1,329.1,198,470.2C1219,174.3,1488.7,226.2,1758.4,139.6z"/></g></g>
</svg>
