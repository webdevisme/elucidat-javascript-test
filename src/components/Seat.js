import React from "react";
import seatIcon from './Seat.svg';
import './Seat.css';

function Seat(props) {
    const handleClick = (event) => {
        event.preventDefault();
        if (props.available) {
            props.seatSelected(props.seatNumber);
        } else {
            props.seatSelected(null);
        }
    }
  return ( props.data ?
    <div className="Seat" available={props.data.available ? 'true' : 'false'}
        onClick={handleClick}>
        <img src={seatIcon}  alt="Seat Icon"/>
        <div className="tooltip">{props.data.seatNumber} : {props.data.available ? 'Available' : ' Taken'}</div>
    </div> 
    : <div className="error">Component Data Error</div>
  );
}

export default Seat;
