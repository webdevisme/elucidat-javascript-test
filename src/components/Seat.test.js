import React from 'react';
import renderer from 'react-test-renderer';
import Seat from './Seat';
import seatsData from '../../public/seats.json';

it('renders seat data passed to prop', () => {
  const component = renderer.create(<Seat data={seatsData[0]}></Seat>);
  
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
