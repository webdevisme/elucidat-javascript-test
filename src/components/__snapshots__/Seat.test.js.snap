// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders seat data passed to prop 1`] = `
<div
  available="true"
  className="Seat"
  onClick={[Function]}
>
  <img
    alt="Seat Icon"
    src="Seat.svg"
  />
  <div
    className="tooltip"
  >
    1A
     : 
    Available
  </div>
</div>
`;
